confluent_kafka/avro/cached_schema_registry_client.py

```diff
@@ -59,32 +60,29 @@ def __init__(self, url, max_schemas_per_subject=1000):
         # subj => { schema => version }
         self.subject_to_schema_versions = defaultdict(dict)
 
-    def _send_request(self, url, method='GET', body=None, headers=None):
-        if body:
-            body = json.dumps(body)
-            body = body.encode('utf8')
-        _headers = dict()
-        _headers["Accept"] = ACCEPT_HDR
-        if body:
-            _headers["Content-Length"] = str(len(body))
-            _headers["Content-Type"] = "application/vnd.schemaregistry.v1+json"
-
-        if headers:
-            for header_name in headers:
-                _headers[header_name] = headers[header_name]
-        if method == 'GET':
-            response = requests.get(url, headers=_headers)
-        elif method == 'POST':
-            response = requests.post(url, body, headers=_headers)
-        elif method == 'PUT':
-            response = requests.put(url, body, headers=_headers)
-        elif method == 'DELETE':
-            response = requests.delete(url, headers=_headers)
-        else:
+        self._session = requests.Session()
+        with self._session as s:
+            if ca_path:
+                s.verify = ca_path
+            if cert_location or key_location:
+                if not key_location and cert_location:
+                    raise ValueError("Both cert_location and key_location must be set: {} {}".format(
+                        cert_location,
+                        key_location))
+                s.cert = (cert_location, key_location)
+
+    def _send_request(self, url, method='GET', body=None, headers={}):
+        if method not in VALID_METHODS:
             raise ClientError("Invalid HTTP request type")
 
-        result = json.loads(response.text)
-        return (result, response.status_code)
+        headers['Accept'] = ACCEPT_HDR
```

Comment:
This modifies the passed headers, use a local _headers copy as the original code.
Suggest adding a test-case that covers this scenario.